llvm/lib/CAS/CASDB.cpp

```diff
@@ -7,17 +7,30 @@
 //===----------------------------------------------------------------------===//
 
 #include "llvm/CAS/CASDB.h"
-#include "llvm/CAS/Utils.h"
-#include "llvm/Support/Allocator.h"
 #include "llvm/Support/Debug.h"
 #include "llvm/Support/FileSystem.h"
-#include "llvm/Support/Path.h"
 
 using namespace llvm;
 using namespace llvm::cas;
 
 LLVM_DUMP_METHOD void CASDB::dump() const { print(dbgs()); }
 
+
+Expected<CASID> CASDB::parseCASID(StringRef Reference) {
+  UniqueID ID;
```

Comment:
This function has a lifetime issue as UniqueID is free'd as function returned.

Reply:
Nice catch; this should be pushing the IDs onto a bumpptrallocator like `BuiltinCAS::parseCASID` was before; I'll fix.